test/robot/search/eval/eval.go

```diff
@@ -141,12 +143,22 @@ func compileEqual(ctx context.Context, expr *search.Binary, t reflect.Type) (eva
 	if err != nil {
 		return nil, boolType, err
 	}
-	if lt != rt {
-		return nil, boolType, log.Err(ctx, nil, "Types to equal do not match")
+	switch {
+	case lt == rt:
+		return func(ctx context.Context, value interface{}) interface{} {
+			return reflect.DeepEqual(lhs(ctx, value), rhs(ctx, value))
+		}, boolType, nil
+	case lt.Implements(stringerType) && rt.Kind() == reflect.String:
```

Comment:
Instead of checking for `stringerType`, you can just use `fmt.Sprint` with the value. It does magic to print the value if it supports [`Stringer`](https://golang.org/pkg/fmt/#Stringer), [`Formater`](https://golang.org/pkg/fmt/#Formatter), and has special cases for most other types.

Reply:
Done.